semgrep/semgrep/evaluation.py

```diff
@@ -134,6 +161,30 @@ def _evaluate_single_expression(
             }
         )
         return output_ranges
+    elif expression.operator == OPERATORS.METAVARIABLE_REGEX:
+        if (
+            not isinstance(expression.operand, dict)
+            or "metavariable" not in expression.operand
+            or "regex" not in expression.operand
+        ):
+            raise SemgrepError(
```

Comment:
is this the right error type? Don't we have an error type that represents invalid (semantically) YAML pattern?

Reply:
Good catch! Are you thinking of `InvalidPatternError`? That seems like the best fit here

Reply:
Actually, `InvalidRuleSchemaError` may be better since this isn't a pattern